lib/stream_data.ex

```diff
@@ -1924,27 +1930,28 @@ defmodule StreamData do
     seed = new_seed(Keyword.fetch!(options, :initial_seed))
     size = Keyword.get(options, :initial_size, 1)
     max_shrinking_steps = Keyword.get(options, :max_shrinking_steps, 100)
-    max_runs = Keyword.get(options, :max_runs, 100)
+    max_runs = Keyword.get(options, :max_runs, :not_set)
+    max_run_time = Keyword.get(options, :max_run_time, :infinity)
+    config = generate_config(max_runs, max_shrinking_steps, max_run_time)
```

Comment:
I would do something like:

```elixir
start_time = System.system_time(:millisecond)
config = %{max_shrinking_steps: max_shrinking_steps}

config =
  case {Keyword.get(options, :max_runs), Keyword.get(options, :max_run_time, :infinity)} do
    {max_runs, :infinity} ->
      Map.put(config, :max_runs, max_runs || 100)
    {nil, max_run_time} ->
      Map.put(config, :max_end_time, start_time + max_run_time)
    {max_runs, max_run_time} ->
      Map.merge(config, %{max_end_time: start_time + max_run_time, max_runs: max_runs}) 
  end

check_all(data, seed, size, fun, _runs = 0, start_time, config)
```

To compress code a little bit. Wdyt?

Reply:
Yeah, works for me. I originally had something like this, but it seemed like a big enough piece of logic to warrant its own function, which is why I extracted it. I'm happy with this if you'd prefer.